src/shared/components/LomFieldsInput/LomFieldsInput.tsx

```diff
@@ -1,126 +1,63 @@
-	const [subjectsOptions, setSubjectsOptions] = useState<Avo.Lom.LomField[]>([]);
-	const [themesOptions, setThemesOptions] = useState<Avo.Lom.LomField[]>([]);
 
-	useEffect(() => {
-		let fetch = true;
-
-		LomService.fetchEducationLevels()
-			.then((res: Avo.Lom.LomField[]) => {
-				fetch && setEducationLevelsOptions(res);
-			})
-			.catch((err) => {
-				console.error(
-					new CustomError('Failed to get education levels from the database', err)
-				);
-			});
-
-		return () => {
-			fetch = false;
-		};
-	}, [setEducationLevelsOptions]);
-
-	useEffect(() => {
-		let fetch = true;
-
-		LomService.fetchSubjects(map(subjects, 'id'))
-			.then((res: Avo.Lom.LomField[]) => {
-				fetch && setSubjectsOptions(res);
-			})
-			.catch((err) => {
-				console.error(new CustomError('Failed to get subjects from the database', err));
-			});
-
-		return () => {
-			fetch = false;
-		};
-	}, [setSubjectsOptions]);
-
-	useEffect(() => {
-		let fetch = true;
-
-		LomService.fetchThemes(map(themes, 'id'))
-			.then((res: Avo.Lom.LomField[]) => {
-				fetch && setThemesOptions(res);
-			})
-			.catch((err) => {
-				console.error(new CustomError('Failed to get themes from the database', err));
-			});
-
-		return () => {
-			fetch = false;
-		};
-	}, [setThemesOptions]);
+	const lomFields = useMemo(() => {
+		return groupLoms(loms as { lom?: Avo.Lom.LomField }[]);
+	}, [loms]);
+	const { data: allEducationLevels, isLoading: isEducationLevelsLoading } =
+		useGetLomEducationLevels();
+	const { data: allSubjects, isLoading: isSubjectsLoading } = useGetLomSubjects(
+		map(lomFields.educationLevels, 'id')
+	);
+	const { data: allThemes, isLoading: isThemesLoading } = useGetLomThemes(
+		map([...lomFields.educationLevels, ...lomFields.subjects], 'id')
+	);
 
-	const handleEducationLevelsChange = (levels: Avo.Lom.LomField[]) => {
-		// onChangeEducationLevels(levels);
-		onChange({ educationLevels: levels });
+	const handleChange = (
+		values: TagInfoSchema[],
+		scheme: LomSchemeType,
+		allSchemeLoms: Avo.Lom.LomField[]
+	) => {
+		const mappedLoms = mapOptionsToLomFields(values, allSchemeLoms);
+		const newLoms = { ...lomFields, [scheme]: mappedLoms };
+		let flatLomList: Avo.Lom.LomField[];
+		flatLomList = Object.values(newLoms).flatMap((lomField) => lomField);
 
-		if (isEmpty(levels)) {
-			onChange({ educationLevels: levels, contexts: [] });
-			return;
-		}
+		if (scheme === 'educationLevels') {
+			const parentContexts = getParentContext(mappedLoms, allEducationLevels || []);
 
-		// Check the difference between the chosen edu levels and the newly picked edu levels arrays to find the added or removed edu level
-		let newEduLevel = differenceWith(levels, educationLevels, isEqual)[0];
-		let added: boolean;
-		if (!isEmpty(newEduLevel)) {
-			added = true;
-		} else {
-			newEduLevel = differenceWith(educationLevels, levels, isEqual)[0];
-			added = false;
+			flatLomList = concat(flatLomList, parentContexts);
```

Comment:
no need to use a lodash function here:
```suggestion
			flatLomList = [...flatLomList, ...parentContexts];
```